src/Sarif.Viewer.VisualStudio/SarifCommandPackage.vsct

```diff
@@ -68,77 +55,11 @@
-        <Parent guid="guidOpenSarifFileCommandPackageCmdSet" id="OpenFileSubMenuGroup" />
-        <Strings>
-          <ButtonText>Import Static Driver Verifier trace log file...</ButtonText>
-        </Strings>
-      </Button>
-      <Button guid="guidOpenSarifFileCommandPackageCmdSet" id="OpenFortifyFileCommandId" priority="0x0140" type="Button">
-        <Parent guid="guidOpenSarifFileCommandPackageCmdSet" id="OpenFileSubMenuGroup" />
-        <Strings>
-          <ButtonText>Import Fortify XML log file...</ButtonText>
-        </Strings>
-      </Button>
-      <Button guid="guidOpenSarifFileCommandPackageCmdSet" id="OpenFortifyFprFileCommandId" priority="0x0150" type="Button">
-        <Parent guid="guidOpenSarifFileCommandPackageCmdSet" id="OpenFileSubMenuGroup" />
-        <Strings>
-          <ButtonText>Import Fortify FPR log file...</ButtonText>
-        </Strings>
-      </Button>
-      <Button guid="guidOpenSarifFileCommandPackageCmdSet" id="OpenClangFileCommandId" priority="0x0160" type="Button">
-        <Parent guid="guidOpenSarifFileCommandPackageCmdSet" id="OpenFileSubMenuGroup" />
-        <Strings>
-          <ButtonText>Import Clang XML log file...</ButtonText>
-        </Strings>
-      </Button>
-      <Button guid="guidOpenSarifFileCommandPackageCmdSet" id="OpenCppCheckFileCommandId" priority="0x0170" type="Button">
-        <Parent guid="guidOpenSarifFileCommandPackageCmdSet" id="OpenFileSubMenuGroup" />
-        <Strings>
-          <ButtonText>Import CppCheck XML log file...</ButtonText>
-        </Strings>
-      </Button>
-      <Button guid="guidOpenSarifFileCommandPackageCmdSet" id="OpenAndroidStudioFileCommandId" priority="0x0180" type="Button">
-        <Parent guid="guidOpenSarifFileCommandPackageCmdSet" id="OpenFileSubMenuGroup" />
-        <Strings>
-          <ButtonText>Import Android Studio XML log file...</ButtonText>
-        </Strings>
-      </Button>
-      <Button guid="guidOpenSarifFileCommandPackageCmdSet" id="OpenSemmleFileCommandId" priority="0x0190" type="Button">
-        <Parent guid="guidOpenSarifFileCommandPackageCmdSet" id="OpenFileSubMenuGroup" />
-        <Strings>
-          <ButtonText>Import Semmle QL CSV log file...</ButtonText>
-        </Strings>
-      </Button>
-      <Button guid="guidOpenSarifFileCommandPackageCmdSet" id="OpenTSLintFileCommandId" priority="0x01A0" type="Button">
-        <Parent guid="guidOpenSarifFileCommandPackageCmdSet" id="OpenFileSubMenuGroup" />
-        <Strings>
-          <ButtonText>Import TSLint log file...</ButtonText>
-        </Strings>
-      </Button>
-      <Button guid="guidOpenSarifFileCommandPackageCmdSet" id="OpenPylintFileCommandId" priority="0x01B0" type="Button">
-        <Parent guid="guidOpenSarifFileCommandPackageCmdSet" id="OpenFileSubMenuGroup" />
-        <Strings>
-          <ButtonText>Import Pylint log file...</ButtonText>
+          <ButtonText>Open Static Analysis Results As SARIF</ButtonText>
```

Comment:
>Open [](start = 22, length = 4)

@larrygolding@comcast.net  - should this say "Import" since it doesn't open the file in the editor?

Reply:
If we think that current functionality is correct, then yes. BUT! I don't think the functionality is correct. I think it should open the resulting (possibly converted) SARIF in the editor, just as File, Open does, and for the same reason: it provides a natural "Close" gesture.

---
In reply to: [506571925](https://github.com/microsoft/sarif-visualstudio-extension/pull/224#discussion_r506571925) [](ancestors = 506571925)